test/forwarddiff.jl

```diff
@@ -54,6 +54,17 @@
         derivative_ε1 = ForwardDiff.derivative(ε1 -> compute_force(ε1, 0.0; metal), 0.0)
         @test norm(derivative_ε1 - derivative_ε1_fd) < 1e-4
     end
+
+    @testset "Using PspUpf" begin
+        Si = ElementPsp(:Si; psp=load_psp(silicon.psp_upf))
+        atoms = [Si, Si]
+
+        derivative_ε1_fd = let ε1 = 1e-5
+            (compute_force(ε1, 0.0; atoms) - compute_force(-ε1, 0.0; atoms)) / 2ε1
+        end
+        derivative_ε1 = ForwardDiff.derivative(ε1 -> compute_force(ε1, 0.0; atoms), 0.0)
+        @test norm(derivative_ε1 - derivative_ε1_fd) < 1e-4
+    end
```

Comment:
This test case is currently failing with a large error of ~0.07 against finite differences:
```
Test Failed at ~/git/DFTK.jl/test/forwarddiff.jl:66

  Expression: norm(derivative_ε1 - derivative_ε1_fd) < 0.0001

   Evaluated: 0.07632223701033393 < 0.0001
```

Reply:
This now works, thanks to #1014! (Error down to ~1e-7)